app/code/Magento/ProductAlert/Controller/Unsubscribe/Stock.php

```diff
@@ -53,13 +81,17 @@ public function execute()
                 throw new NoSuchEntityException();
```

Comment:
Please move product retrieval and exception throwing to a new method - to avoid throw and catch in the same method.

Reply:
Done.

Reply:
Cool.